smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQMessageSender.java

```diff
@@ -0,0 +1,272 @@
+     */
+    @Override
+    public void subscribe(
+            final Subscriber<? super Message<?>> subscriber) {
+        if (!downstream.compareAndSet(null, subscriber)) {
+            Subscriptions.fail(subscriber, RabbitMQExceptions.ex.illegalStateOnlyOneSubscriberAllowed());
+        } else {
+            if (upstream.get() != null) {
+                subscriber.onSubscribe(this);
+            }
+        }
+    }
+
+    /* ----------------------------------------------------- */
+    /* METHODS OF SUBSCRIBER */
+    /* ----------------------------------------------------- */
+
+    /**
+     * Invoked after calling {@link Publisher#subscribe(Subscriber)}.
+     * <p>
+     * No data will start flowing until {@link Subscription#request(long)} is invoked.
+     * <p>
+     * It is the responsibility of this {@link Subscriber} instance to call {@link Subscription#request(long)} whenever more
+     * data is wanted.
+     * <p>
+     * The {@link Publisher} will send notifications only in response to {@link Subscription#request(long)}.
+     *
+     * @param subscription
+     *        {@link Subscription} that allows requesting data via {@link Subscription#request(long)}
+     */
+    @Override
+    public void onSubscribe(Subscription subscription) {
+        if (this.upstream.compareAndSet(null, subscription)) {
+            Subscriber<? super Message<?>> subscriber = downstream.get();
+            if (subscriber != null) {
+                subscriber.onSubscribe(this);
+            }
+        } else {
+            Subscriber<? super Message<?>> subscriber = downstream.get();
+            if (subscriber != null) {
+                subscriber.onSubscribe(Subscriptions.CANCELLED);
+            }
+        }
+    }
+
+    /**
+     * Data notification sent by the {@link Publisher} in response to requests to {@link Subscription#request(long)}.
+     *
+     * @param message the element signaled
+     */
+    @Override
+    public void onNext(Message<?> message) {
+        if (isCancelled()) {
+            return;
+        }
+
+        final Subscriber<? super Message<?>> subscriber = this.downstream.get();
+
+        retrieveSender
+                .onItem().transformToUni(sender -> {
+                    try {
+                        return send(sender, message, configuredExchange, configuration)
+                                .onItem().transform(m -> Tuple2.of(sender, m));
+                    } catch (Exception e) {
+                        // Message can't be sent - nacking and skipping.
+                        message.nack(e);
+                        RabbitMQLogging.log.serializationFailure(configuration.getChannel(), e);
+                        return Uni.createFrom().nullItem();
+                    }
+                })
+                .subscribe().with(
+                        tuple -> {
+                            if (tuple != null) { // Serialization issue
```

Comment:
Why this comment?

Reply:
It's "inherited" from the identical code in the `AmqpCreditBasedSender.java` file from the AMQP connector; I don't know what the issue is, but I thought it prudent to include it just in case.

Reply:
I believe it's a left-over, you can remove it.